removefiles/activity.go

```diff
@@ -58,21 +59,21 @@ func (a *Activity) Execute(ctx context.Context, params *ActivityParams) (*Activi
 		"Path", params.Path,
 	)
 
-	count, err := removeByNames(params.Path, a.cfg.RemoveNames)
+	count, err := remove(params.Path, a.cfg.RemoveNames, a.cfg.RemovePatterns)
 	if err != nil {
 		return nil, fmt.Errorf("RemoveFilesActivity: %v", err)
 	}
 
 	return &ActivityResult{Count: count}, nil
 }
 
-// removeByNames deletes any file or directory in dir (and sub-directories)
-// whose name matches one of the values in remove, then returns a count of
-// deleted items.
-func removeByNames(dir, names string) (int, error) {
+// remove deletes any file or directory in dir (and sub-directories) whose
+// name matches one of the values in names and/or patterns, then returns a
+// count of deleted items.
+func remove(dir, names, patterns string) (int, error) {
```

Comment:
I think it would be better to keep the `removeByNames()` function and add a new `removeByPatterns()` function for the regex matching, and a general `remove` function to call the first two.  `removeByName()` would be no-op if `names == ""` and `removeByPatterns()` if `patterns == ""`.

Reply:
But if both are set it will walk through the directory twice, I think it's better to avoid that.

Reply:
> But if both are set it will walk through the directory twice, I think it's better to avoid that.

Right, agreed.